esp/esdllib/esdl_script.cpp

```diff
@@ -1402,7 +1457,15 @@ class CEsdlTransformOperationTrace : public CEsdlTransformOperationWithoutChildr
         {
             if (m_test && !sourceContext->evaluateAsBoolean(m_test))
                 return false;
-            targetContext->trace(m_label, m_select, scriptContext->getTraceToStdout());
```

Comment:
I see how you've moved the state variables to the CEsdlScriptContext, but I don't recall how they hook in to a change in behavior.

Reply:
This PR is more about clarifying and enforcing a separation of responsibilities than it is about changing behavior. The change from `trace` to `selectText` separates XML document data from the reason it is being requested.

Should the `trace` ESDL script operation rely on a generic XML interface to implement its specialized output behavior, or should the operation use the XML interface to obtain the data that it needs to manipulate?

Reply:
I think what I meant was that the LOG statements here don't honor the traceToStdOut tag, but should they?

Reply:
Yeah, they should. In the bigger picture, where this originated, the flag is not needed here because it's handled before we ever get here. I guess I need to rework this so I can immediately rip it out in the next PR.